src/NHibernate.Test/CacheTest/QueryCacheFixture.cs

```diff
@@ -60,5 +60,19 @@ public void QueryCacheWithNullParameters()
 				s.Flush();
 			}
 		}
+
+		[Test]
+		public void QueryCacheWithScalarFunctionThatReturnsInt()
+		{
+			using (ISession s = OpenSession())
+			{
+				int result = s
+					.CreateSQLQuery("SELECT [dbo].[TestQueryCache] ()")
```

Comment:
The TestQueryCache method is not in the test database. Use a simple "SELECT 1" or similar.

Reply:
Hi

I will change test case so it will be  testable.

Regards

Reply:
Hi @gliljas 

I have made new feature branch https://github.com/joesun99/nhibernate-core/tree/SecondLevelQueryCacheFixOld 
Should I make new pull request and close current?

Thx

Regards

Reply:
Or update the current PR, unless that is two different approaches and you prefer keep the old one in case it finally should be chosen.